roles/os_net_setup/tasks/main.yml

```diff
@@ -1,53 +1,87 @@
 ---
-- name: Get openstack admin credentials from k8s
-  block:
-    - name: Get clouds.yaml configuration
-      kubernetes.core.k8s_info:
-        kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
-        context: "{{ cifmw_openshift_context }}"
-        kind: ConfigMap
-        namespace: openstack
-        name: openstack-config
-      register: clouds_yaml
-
-    - name: Get the OSP admin password
-      kubernetes.core.k8s_info:
-        kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
-        context: "{{ cifmw_openshift_context }}"
-        kind: Secret
-        namespace: openstack
-        name: openstack-config-secret
-      register: osp_secret
 
-- name: Construct auth param for openstack
+- name: Delete existing resources
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  when: not cifmw_os_net_setup_dry_run | bool
   block:
-    - name: Set openstack auth fact
-      ansible.builtin.set_fact: # noqa: yaml[line-length]
-        openstack_auth: |
-          {{ ((clouds_yaml.resources[0].data['clouds.yaml']| from_yaml)['clouds']['default'] |
-          combine ({'auth': { 'password' : (osp_secret.resources[0].data['secure.yaml']|b64decode|from_yaml)['clouds']['default']['auth']['password']} }, recursive=true))['auth'] }}
-        region_name: "{{ (clouds_yaml.resources[0].data['clouds.yaml']| from_yaml)['clouds']['default']['region_name'] }}"
-  rescue:
-    - name: Fail if we do not have auth information
-      ansible.builtin.fail:
-        msg: |
-          Failed to Retrive auth Infomation
-          clouds.yaml: {{ clouds_yaml }}
-          osp_secret: {{ osp_secret }}
+    - name: Delete existing subnet pools
```

Comment:
still I think that this may fail cleaning as if there are subnets created from the existing pool, such pool will not be possible to be deleted. I think You should first delete subnets, then subnet pools and networks (order of networks and subnet pools don't really matters, important is that there will be no subnets using subnetpools before deletion of the subnet pools).

Reply:
ack! Fixed!